src/Microsoft.SqlTools.SqlCore/Scripting/ScriptAsScriptingOperation.cs

```diff
@@ -53,6 +53,12 @@ public ScriptAsScriptingOperation(ScriptingParams parameters, string azureAccoun
             }
 
             ServerConnection = new ServerConnection(sqlConnection);
+            if (string.IsNullOrEmpty(ServerConnection.Password)) {
+                // Manually add password as server connection class does not add it to string, required for successful connection.
+                if(ServerConnection.ConnectionString.IndexOf("Authentication=SqlPassword") != -1 && ServerConnection.ConnectionString.IndexOf("Password=") == -1) {
```

Comment:
Please verify connection string where SqlPassword authentication mode is not specified, as it's auto determined by the driver based on username specified.

Correct logic should be to only check if user is specified, and password is not, you can set password to be empty - it will work or fail based on server authentication requirements.

Use SqlConnectionStringBuilder to compare, as mentioned before.